src/lib/homestore.cpp

```diff
@@ -224,13 +224,33 @@ void HomeStore::format_and_start(std::map< uint32_t, hs_format_params >&& format
             m_index_service->create_vdev(pct_to_size(fparams.size_pct, fparams.dev_type), fparams.dev_type,
                                          fparams.num_chunks);
         } else if ((svc_type & HS_SERVICE::DATA) && has_data_service()) {
-            m_data_service->create_vdev(pct_to_size(fparams.size_pct, fparams.dev_type), fparams.dev_type,
-                                        fparams.block_size, fparams.alloc_type, fparams.chunk_sel_type,
-                                        fparams.num_chunks);
+            if (fparams.num_chunks != 0) {
+                m_data_service->create_vdev_with_num_chunks(pct_to_size(fparams.size_pct, fparams.dev_type),
+                                                            fparams.dev_type, fparams.block_size, fparams.alloc_type,
+                                                            fparams.chunk_sel_type,
+                                                            fparams.num_chunks);
+            } else if (fparams.chunk_size != 0) {
```

Comment:
when creating vdev, device_manager always prioritize num_chunks over chunk_size
https://github.com/eBay/HomeStore/blob/6a2dfd8e212a9a4b5c9b17c3d1de45acb3907ce2/src/lib/device/device_manager.cpp#L243-L244

if we want to create vdev according to num_chunks , we set it to non-zero, otherwist, we set num_chunks to zero ,and set chunk_size to what we want

Reply:
good idea